app/code/Magento/Sales/Observer/SalesEventQuoteSubmitBeforeObserver.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Sales\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+
+/**
+ * Copy custom fields from quote tables to order.
+ */
+class SalesEventQuoteSubmitBeforeObserver implements ObserverInterface
+{
+    /**
+     * @var \Magento\Framework\DataObject\Copy\Config
+     */
+    private $fieldsetConfig;
+
+    /**
+     * @var \Magento\Sales\Api\Data\OrderInterface
+     */
+    private $order;
+
+    /**
+     * SalesEventQuoteSubmitBeforeObserver constructor.
+     *
+     * @param \Magento\Framework\DataObject\Copy\Config $fieldsetConfig
+     * @param \Magento\Sales\Api\Data\OrderInterface $order
+     */
+    public function __construct(
+        \Magento\Framework\DataObject\Copy\Config $fieldsetConfig,
+        \Magento\Sales\Api\Data\OrderInterface $order
+    ) {
+        $this->fieldsetConfig = $fieldsetConfig;
+        $this->order = $order;
+    }
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     *
+     * @return $this
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        /** @var \Magento\Quote\Api\Data\CartInterface $quote */
+        $quote = $observer->getEvent()->getQuote();
+
+        /** @var \Magento\Sales\Api\Data\OrderInterface $order */
+        $order = $observer->getEvent()->getOrder();
+
+        $fields = $this->fieldsetConfig->getFieldset('sales_convert_quote', 'global');
+
+        $methods = get_class_methods($this->order);
```

Comment:
Why do we need this if `$order` is `\Magento\Sales\Api\Data\OrderInterface`?

Reply:
Yes, this is my fault. Fixed.